kedro/io/partitioned_data_set.py

```diff
@@ -97,8 +102,9 @@ def __init__(  # pylint: disable=too-many-arguments
         dataset: Union[str, Type[AbstractDataSet], Dict[str, Any]],
         filepath_arg: str = "filepath",
         filename_suffix: str = "",
-        credentials: Dict[str, Any] = None,
```

Comment:
Sorry, `credentials` need to stay right after the `filename_suffix` not to break the order of the arguments in the constructor. To keep things 100% backwards compatible when extending the arguments, we should only add new args to the end of the list (and of course keep then optional).

Reply:
Let me push a quick commit for that